infra/main.tf

```diff
@@ -0,0 +1,595 @@
+    "key.json" = base64decode(google_service_account_key.monitoring_key.private_key)
+  }
+}
+
+resource "random_id" "test_name_suffix" {
+  byte_length = 2
+}
+
+resource "google_service_account" "test" {
+  account_id = "test-${random_id.test_name_suffix.hex}"
+}
+
+resource "google_service_account_key" "test_key" {
+  service_account_id = google_service_account.test.name
+}
+
+resource "kubernetes_secret" "test_gsa_key" {
+  metadata {
+    name = "test-gsa-key"
+  }
+
+  data = {
+    "key.json" = base64decode(google_service_account_key.test_key.private_key)
+  }
+}
+
+resource "google_project_iam_member" "test_compute_instance_admin" {
+  role = "roles/compute.instanceAdmin.v1"
+  member = "serviceAccount:${google_service_account.test.email}"
+}
+
+resource "google_project_iam_member" "test_service_account_user" {
+  role = "roles/iam.serviceAccountUser"
+  member = "serviceAccount:${google_service_account.test.email}"
+}
+
+resource "google_project_iam_member" "test_logging_viewer" {
+  role = "roles/logging.viewer"
+  member = "serviceAccount:${google_service_account.test.email}"
+}
+
+resource "google_project_iam_member" "test_service_usage_consumer" {
+  role = "roles/serviceusage.serviceUsageConsumer"
+  member = "serviceAccount:${google_service_account.test.email}"
+}
+
+resource "google_service_account" "test_dev" {
+  account_id = "test-dev"
+}
+
+resource "google_service_account_key" "test_dev_key" {
+  service_account_id = google_service_account.test_dev.name
+}
+
+resource "kubernetes_secret" "test_dev_gsa_key" {
+  metadata {
+    name = "test-dev-gsa-key"
+  }
+
+  data = {
+    "key.json" = base64decode(google_service_account_key.test_dev_key.private_key)
+  }
+}
+
+resource "google_service_account" "batch_agent" {
+  account_id = "batch2-agent"
+}
+
+resource "google_project_iam_member" "batch_agent_compute_instance_admin" {
+  role = "roles/compute.instanceAdmin.v1"
+  member = "serviceAccount:${google_service_account.batch_agent.email}"
+}
+
+resource "google_project_iam_member" "batch_agent_service_account_user" {
+  role = "roles/iam.serviceAccountUser"
+  member = "serviceAccount:${google_service_account.batch_agent.email}"
+}
+
+resource "google_project_iam_member" "batch_agent_log_writer" {
+  role = "roles/logging.logWriter"
+  member = "serviceAccount:${google_service_account.batch_agent.email}"
+}
+
+resource "google_project_iam_member" "batch_agent_object_creator" {
+  role = "roles/storage.objectCreator"
+  member = "serviceAccount:${google_service_account.batch_agent.email}"
+}
+
+resource "google_project_iam_member" "batch_agent_object_viewer" {
+  role = "roles/storage.objectViewer"
+  member = "serviceAccount:${google_service_account.batch_agent.email}"
+}
+
+resource "google_compute_firewall" "vdc_to_batch_worker" {
+  name    = "vdc-to-batch-worker"
+  network = google_compute_network.internal.name
+
+  source_ranges = [google_container_cluster.vdc.cluster_ipv4_cidr]
+
+  target_tags = ["batch2-worker"]
```

Comment:
Where is this network tag set?

Should this rule apply to the default network?